homeassistant/components/dsmr/sensor.py

```diff
@@ -286,12 +288,19 @@ def icon(self):
     def state(self):
         """Return the state of sensor, if available, translate if needed."""
         value = self.get_dsmr_object_attr("value")
+        unit = self.get_dsmr_object_attr("unit")
 
         if self._obis == obis_ref.ELECTRICITY_ACTIVE_TARIFF:
             return self.translate_tariff(value, self._config[CONF_DSMR_VERSION])
 
         try:
-            value = round(float(value), self._config[CONF_PRECISION])
+            if unit == POWER_KILO_WATT:
+                value = round(
+                    float(value) * 1000.0,
+                    self._config[CONF_PRECISION],
+                )
```

Comment:
We should not change/calculate units from Home Assistant. Instead, Home Assistant should provide the units of measurement from upstream as-is.

Reply:
It was not an issue here #37896, that merged not so long ago.

Reply:
Right, still, we don't want to avoid this happening inside ha.

Reply:
And if we can't avoid it? Solving it upstream is not straightforward as it doesn't care about the content of the data and it is nearly impossible without breaking tons of things.

Seems to me there is not really a strategy here. There are other integrations converting data (like rssi in rfxtrx integration), there is a clear ADR that communication should be done by a third party library but to me it's not clear that no data may be converted inside ha.

Reply:
So, the change here is kinda personal preference. I agree, I use Watt in my home as well. However, we are now changing the value and unit of measurement of the sensor. The only reasoning given is because: "watts are more common".

However, the repercussion is: Introducing a breaking change for everybody relying on this already, for a reason that is purely a cosmetic/opinionated thing. People need to adjust their automations, which will probably end up with a broken history recording (internal/external recorded stored data), and a change measurement unit (which affects e.g., InfluxDB data).

IMHO, we should not make breaking changes like that purely for the sake of the given reasoning.

Reply:
On that point I can follow you. I was doubting whether to make a breaking change and kind of followed #37896.

It's not purely cosmetics, these values often end up in a database and in charts. Would be helpful to have one unit.

What if we make it an option?

Reply:
I'm not a fan of options 😄 
Maybe add another sensor, make the Kw version disabled by default (so existing users are not affected)? (thinking out loud now).

Reply:
I'm neutral to having options but open for ideas.
Seems like an elegant way to avoid options and breaking changes. I'll adjust it to this.

Reply:
Did you mean make W default disabled? Or kW?

Reply:
I would say, disable kW by default. As W makes more sense, but disabling by default doesn't effect existing users (as the sensor is already enabled).